src/Draft.js

```diff
@@ -71,6 +72,7 @@ const DraftPublic = {
   genKey: generateRandomKey,
   getDefaultKeyBinding,
   getVisibleSelectionRect,
+  getDraftEditorSelection,
```

Comment:
What if, instead of exposing this, we call `update` and pass in an `EditorState` with the selection we want?

Alternatively, would [this PR](https://github.com/facebook/draft-js/pull/1515) help?

Reply:
We get a mousedown event with coordinates on the screen that we can translate into a selection, but just setting that selection hasn't been enough because draft frequently ignores it. As you say, it works a lot better to call update with an `EditorState` that has the correct `SelectionState`, but we need this function to do that. Without it, we have no way to translate a conventional selection intro a draft selection in the first place.

That PR wouldn't replace the need to do this. We do have race conditions here, because the mousedown event that we want to move the selection to also triggers a `focusContent` action, but I've gotten around those problems for now with a `setTimeout` set to 0. That way we only try to correct the selection once draft has finished. It's still something we should consider though--the bug that PR addresses likely contributed to how difficult the EOE statement stuff was.

Reply:
Why don't we know where we want the selection to go (I may be misunderstanding the bug this is trying to correct - I thought we already knew the selection state but some other issue was resetting that to a position we didn't want)?

Reply:
That's mostly correct, but it's a little hard to talk about because we're calling everything a selection. The only thing that's no longer really true is that the part about resetting it to a position we don't want--we got around that by letting draft do whatever it wants, then setting the selection to whatever we want it to be in a `setTimeout`. At that point though, we're setting the selection based on the coordinates of a mousedown event, and we're translating that into a normal selection, not a draft `SelectionState`. 

That works, but the problem now is that draft may ignore the fact that we've updated the selection. As a result, subsequent changes to the text happen where Draft thinks the selection is, not where we've set the selection to. Exporting this lets us parse any selection we set into a `SelectionState` so that we can update draft ourselves instead of relying on Draft to notice the selection change.

Reply:
I'm going to admit I'm not super satisfied with this change because every change to Draft is a vast maintenance burden. I believe that there must be a better way but the investigation there is likely far more than its worth right now. The comment in the frontend change should be enough to trace this problem back and I think there's enough of a paper trail elsewhere for us to confidently remove this should we ever find a better solution.